src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -544,7 +549,7 @@ partial class BoundBaseReference : IInstanceReferenceExpression
 
     partial class BoundThisReference : IInstanceReferenceExpression
     {
-        bool IInstanceReferenceExpression.IsExplicit => true;
+        bool IInstanceReferenceExpression.IsExplicit => this.Syntax.IsKind(SyntaxKind.ThisExpression);
```

Comment:
The existing implementation was of course completely incorrect. My guess is that I noticed that there wasn't a convenient property to test (as there is in the VB bound tree representation), coded a bogus implementation and moved on. Just out of curiosity, what is the representation of an implicit "this"?